src/ports/postgres/modules/deep_learning/test/input_data_preprocessor.sql_in

```diff
@@ -88,6 +89,23 @@ SELECT training_preprocessor_dl(
   'label',
   'x');
 
+-- Test data is evenly distributed across all segments (GPDB only)
+m4_changequote(`<!', `!>')
+m4_ifdef(<!__POSTGRESQL__!>, <!!>, <!
+DROP TABLE IF EXISTS data_preprocessor_input_batch, data_preprocessor_input_batch_summary;
+SELECT training_preprocessor_dl(
+  'data_preprocessor_input',
+  'data_preprocessor_input_batch',
+  'id',
+  'x',
+  1);
+
+SELECT assert(count(*)=(SELECT ceil(17.0/count(*)) from gp_segment_configuration WHERE role = 'p' and content != -1), 'Even distribution of buffers failed.')
+FROM data_preprocessor_input_batch
+WHERE gp_segment_id = 0
+GROUP BY gp_segment_id;
```

Comment:
If we have the `WHERE gp_segment_id = 0` clause, why do we need the `group by`?

Reply:
Good catch! Initially we were planning to count images for all the segments, but now we only want to assert on a single segment so `GROUP BY ` doesn't make sense. Will update the test query to remove it.